packages/router/router.ts

```diff
@@ -343,7 +344,11 @@ export interface RouterInit {
   routes: AgnosticRouteObject[];
   history: History;
   basename?: string;
+  /**
+   * @deprecated Use `enhanceAgnosticRoute` instead
+   */
   detectErrorBoundary?: DetectErrorBoundaryFunction;
+  enhanceAgnosticRoute?: EnhanceAgnosticRouteFunction;
```

Comment:
I feel like this name could be clearer, e.g. `resolveRouteProperties`?

Reply:
Definitely open to bike-shedding here.  I kinda like indicating that this function is what takes us from framework-agnostic to framework-aware.  `resolveFrameworkAwareRouteProperties` is a mouthful though 😕 .  `resolveRouteFrameworkProperties` or `resolveRouteRenderProperties` maybe?  

Do you think the word "resolve" carries an incorrect connotation that this is how `route.lazy` stuff gets applied?  That's not quite correct since this is a second pass after we apply the `route.lazy` props.

Longer term, I could even see a future where we decide that `Component/ErrorBoundary` are framework-agnostic enough (in todays component-based framework landscape, and as opposed to `element`/`errorElement` which felt super react-specific) and maybe we just type them as `any` in the router and that would eliminate `hasErrorBoundary` entirely.  Then we overload the types in the router layer with `React.Component` etc.

Reply:
This could also be a future flag, but it's an implementation detail of RR and only newly (6.9.0) public API in `@remix-run/router` so on the fence as to whether it's worth a formal future flag?

Reply:
Bikeshedded and currently thinking `mapRouteProperties`